src/compiler/checker.ts

```diff
@@ -17604,7 +17604,9 @@ namespace ts {
             // and the right operand to be of type Any, a subtype of the 'Function' interface type, or have a call or construct signature.
             // The result is always of the Boolean primitive type.
             // NOTE: do not raise error if leftType is unknown as related error was already reported
-            if (!isTypeAny(leftType) && isTypeAssignableToKind(leftType, TypeFlags.Primitive)) {
+            if (!isTypeAny(leftType) &&
+                (isTypeAssignableToKind(leftType, TypeFlags.Primitive) ||

```

Comment:
I wonder if it makes sense to hae `isTypeAssignableToKind` just work recursively for all constituents if given a union type. Perhaps make an `allTypesAssignableToKind` helper below it to keep the change scoped.

Otherwise, this seems fine.